components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TabListReducer.kt

```diff
@@ -18,8 +18,23 @@ internal object TabListReducer {
     fun reduce(state: BrowserState, action: TabListAction): BrowserState {
         return when (action) {
             is TabListAction.AddTabAction -> {
+
+                val updatedTabList = if (action.parent != null) {
```

Comment:
Since the logic for adding a tab from a parent is fairly different, should it be its own action?

Reply:
I am not sure. They are quite similar, minus the one if statement and I like to keep the API symmetric AddTab/RemoveTab vs. AddTab/AddChildTab/RemoveTab. When removing, the caller shouldn't need to worry about whether a tab is a parent/child tab or not.

Reply:
Happy to change though if we feel it's getting too complicated? Can also be refactored later though...